### PR TITLE
apiResponse: FIXES routes updates and devices api responses.

### PR DESCRIPTION
**Pull request by ldjebran**
FIXES routes updates and devices api responses, using appropriate functions RespondWithAPIError and respondWithJSONBody.

## Type of change

What is it?

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [X] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

**Review by omaciel — CHANGES_REQUESTED** [review]
Would love for existing tests to also check for the type of response: `respondWithAPIError` and `respondWithJSONBody`.

**Comment by ldjebran** [comment]
@omaciel we did not change the type of response , but used the compact version of it via functions respondWithAPIError and respondWithJSONBody  the code became more readable, and we removed repetitive code .

**Comment by omaciel** [comment]
> @omaciel we did not change the type of response , but used the compact version of it via functions respondWithAPIError and respondWithJSONBody the code became more readable, and we removed repetitive code .

I understood that bit :) I'm proposing that we also test the response type so that if anyone ever decides or changes it by mistake in the future, our tests will alert us.

**Comment by omaciel** [comment]
@ldjebran and I discussed over Slack and agreed that what I suggested requires a much more involved code refactoring, so I'll mark this as resolved.

File: pkg/routes/devices.go [hunk 1]
```diff
@@ -97,6 +96,7 @@ var devicesFilters = common.ComposeFilters(
 // ValidateGetAllDevicesFilterParams validate the query params that sent to /devices endpoint
 func ValidateGetAllDevicesFilterParams(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctxServices := dependencies.ServicesFromContext(r.Context())
```
Comment (omaciel) [inline pkg/routes/devices.go:99]: Please consider adding at least one single unittest for this function? We currently have zero coverage:
Screenshot 2022-11-23 at 10 23 01

File: pkg/routes/updates.go [hunk 2]
```diff
@@ -91,22 +90,14 @@ func GetUpdatePlaybook(w http.ResponseWriter, r *http.Request) {
 	playbook, err := services.UpdateService.GetUpdatePlaybook(update)
 	if err != nil {
 		services.Log.WithField("error", err.Error()).Error("Error getting update playbook")
-		err := errors.NewNotFound("file was not found on the S3 bucket")
-		w.WriteHeader(err.GetStatus())
-		if err := json.NewEncoder(w).Encode(&err); err != nil {
-			services.Log.WithField("error", err.Error()).Error("Error while trying to encode")
-		}
+		respondWithAPIError(w, services.Log, errors.NewNotFound("file was not found on the S3 bucket"))
```
Comment (omaciel) [inline pkg/routes/updates.go:93]: Please consider adding at least one unittest for this function as we currently have zero coverage:


Screenshot 2022-11-23 at 10 26 20

File: pkg/routes/updates.go [hunk 3]
```diff
@@ -126,17 +117,11 @@ func GetUpdates(w http.ResponseWriter, r *http.Request) {
 		services.Log.WithFields(log.Fields{
 			"error": result.Error.Error(),
 		}).Error("Error retrieving updates")
-		err := errors.NewInternalServerError()
-		w.WriteHeader(err.GetStatus())
-		if err := json.NewEncoder(w).Encode(&err); err != nil {
-			services.Log.WithField("error", err.Error()).Error("Error while trying to encode")
-		}
+		respondWithAPIError(w, services.Log, errors.NewInternalServerError())
```
Comment (omaciel) [inline pkg/routes/updates.go:120]: We currently have zero coverage for this function:

Screenshot 2022-11-23 at 10 27 52

File: pkg/routes/updates.go [hunk 4]
```diff
@@ -291,15 +276,13 @@ func AddUpdate(w http.ResponseWriter, r *http.Request) {
 
 // GetUpdateByID obtains an update from the database for an orgID
 func GetUpdateByID(w http.ResponseWriter, r *http.Request) {
+	ctxServices := dependencies.ServicesFromContext(r.Context())
```
Comment (omaciel) [inline pkg/routes/updates.go:279]: Zero coverage:

Screenshot 2022-11-23 at 10 29 06

File: pkg/routes/updates.go [hunk 5]
```diff
@@ -315,23 +298,19 @@ func getUpdate(w http.ResponseWriter, r *http.Request) *models.UpdateTransaction
 // SendNotificationForDevice TMP route to validate
 func SendNotificationForDevice(w http.ResponseWriter, r *http.Request) {
 	if update := getUpdate(w, r); update != nil {
-		services := dependencies.ServicesFromContext(r.Context())
-		notify, err := services.UpdateService.SendDeviceNotification(update)
+		ctxServices := dependencies.ServicesFromContext(r.Context())
```
Comment (omaciel) [inline pkg/routes/updates.go:301]: Zero coverage:

Screenshot 2022-11-23 at 10 30 12

File: pkg/routes/updates.go [hunk 6]
```diff
@@ -394,6 +373,7 @@ var updateFilters = common.ComposeFilters(
 // ValidateGetUpdatesFilterParams validate the query params that sent to /updates endpoint
 func ValidateGetUpdatesFilterParams(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctxServices := dependencies.ServicesFromContext(r.Context())
```
Comment (omaciel) [inline pkg/routes/updates.go:376]: Zero coverage:

Screenshot 2022-11-23 at 10 31 26

**Review by omaciel — CHANGES_REQUESTED** [review]
Please consider adding at least one single and comprehensive test for the functions I highlighted. I see these as serious technical debts for us.

File: pkg/routes/devices.go (hunk 1 above)
Comment (ldjebran) [reply pkg/routes/devices.go:99]: image

Comment (ldjebran) [reply pkg/routes/devices.go:99]: done

File: pkg/routes/updates.go (hunk 2 above)
Comment (ldjebran) [reply pkg/routes/updates.go:93]: image
The io.Copy error needs too much code refactoring, will do it in a separate PR asn io copy needs to be transfred via  context ir try to use file service

Comment (ldjebran) [reply pkg/routes/updates.go:93]: done

File: pkg/routes/updates.go (hunk 3 above)
Comment (ldjebran) [reply pkg/routes/updates.go:120]: image
the db error need to be investigated as we need to mock db will work with @AaronH88 to find a solution for this.

Comment (ldjebran) [reply pkg/routes/updates.go:120]: done

File: pkg/routes/updates.go (hunk 4 above)
Comment (ldjebran) [reply pkg/routes/updates.go:279]: image

Comment (ldjebran) [reply pkg/routes/updates.go:279]: done

File: pkg/routes/updates.go (hunk 5 above)
Comment (ldjebran) [reply pkg/routes/updates.go:302]: image

Comment (ldjebran) [reply pkg/routes/updates.go:302]: done

File: pkg/routes/updates.go (hunk 6 above)
Comment (ldjebran) [reply pkg/routes/updates.go:377]: image

Comment (ldjebran) [reply pkg/routes/updates.go:377]: done

**Comment by ldjebran** [comment]
@omaciel please revisit 
increased coverage:
 for devices from 41.1% upto  50.3
 for updates from 43.9 upto 78%

**Review by omaciel — APPROVED** [review]
/lgtm

**Comment by omaciel** [comment]
From https://app.codecov.io/gh/RedHatInsights/edge-api/pull/1683:

Screenshot 2022-11-26 at 11 35 44

**Comment by ldjebran** [comment]
strange that the numbers are different from the local dev one 
image

image

**Review by mgold1234 — APPROVED** [review]
/lgtm

**Pull request merged**